### PR TITLE
[ome_session - auth and idrac_session - auth] Updating utils and modules to accept x_auth_token as a param

### PR DESCRIPTION
**Pull request by ShivamSh3**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

File: plugins/module_utils/idrac_redfish.py [hunk 1]
```diff
@@ -180,7 +183,7 @@ def invoke_request(self, uri, method, data=None, query_param=None, headers=None,
 
     def __enter__(self):
         """Creates sessions by passing it to header"""
-        if self.req_session:
+        if self.req_session and not self.x_auth_token:
```
Comment (ABHISHEK-SINHA10) [inline plugins/module_utils/idrac_redfish.py:187]: Make it much easier to understand like below:

```
if self.req_session:
     if self.x_auth_token:
           take token in header
     else:
           take username and password in payload
```

File: plugins/doc_fragments/idrac_x_auth_options.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+
+#
+# Dell OpenManage Ansible Modules
+# Version 9.3.0
+# Copyright (C) 2020-2024 Dell Inc. or its subsidiaries. All Rights Reserved.
```
Comment (rajshekarp87) [inline plugins/doc_fragments/idrac_x_auth_options.py]:
```suggestion
# Copyright (C) 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
```

File: plugins/module_utils/idrac_redfish.py [hunk 3]
```diff
@@ -41,14 +41,16 @@
 
 idrac_auth_params = {
     "idrac_ip": {"required": True, "type": 'str'},
-    "idrac_user": {"required": True, "type": 'str', "fallback": (env_fallback, ['IDRAC_USERNAME'])},
-    "idrac_password": {"required": True, "type": 'str', "aliases": ['idrac_pwd'], "no_log": True, "fallback": (env_fallback, ['IDRAC_PASSWORD'])},
+    "idrac_user": {"required": False, "type": 'str', "fallback": (env_fallback, ['IDRAC_USERNAME'])},
+    "idrac_password": {"required": False, "type": 'str', "aliases": ['idrac_pwd'], "no_log": True, "fallback": (env_fallback, ['IDRAC_PASSWORD'])},
+    "x_auth_token": {"required": False, "type": 'str', "no_log": True, "fallback": (env_fallback, ['X_AUTH_TOKEN'])},
```
Comment (rajshekarp87) [inline plugins/module_utils/idrac_redfish.py]:
```suggestion
    "x_auth_token": {"required": False, "type": 'str', "no_log": True, "fallback": (env_fallback, ['IDRAC_X_AUTH_TOKEN'])},
```

File: plugins/module_utils/ome.py [hunk 4]
```diff
@@ -191,18 +196,20 @@ def invoke_request(self, method, path, data=None, query_param=None, headers=None
...
     def __enter__(self):
         """Creates sessions by passing it to header"""
-        if self.req_session:
+        if self.req_session and not self.x_auth_token:
             payload = {'UserName': self.username,
                        'Password': self.password,
                        'SessionType': 'API', }
             path = SESSION_RESOURCE_COLLECTION["SESSION"]
-            resp = self.invoke_request('POST', path, data=payload)
+            resp = self.invoke_request(path, 'POST', data=payload)
```
Comment (ABHISHEK-SINHA10) [inline plugins/module_utils/ome.py]: place of passed parameter seems not correct, 

In utils file it's like "invoke_request(self, method, path,"   ---> method and then path, but new changes is path and then method.

reconfirm once from your side.

Comment (ShivamSh3) [reply plugins/module_utils/ome.py]: Good observation. I don't know how it got changed.

**Review by Kritika-Bhateja-03 — DISMISSED** [review]

File: plugins/module_utils/idrac_redfish.py [hunk 5]
```diff
@@ -41,14 +41,16 @@
...
+    "idrac_user": {"required": False, "type": 'str', "fallback": (env_fallback, ['IDRAC_USERNAME'])},
+    "idrac_password": {"required": False, "type": 'str', "aliases": ['idrac_pwd'], "no_log": True, "fallback": (env_fallback, ['IDRAC_PASSWORD'])},
+    "x_auth_token": {"required": False, "type": 'str', "no_log": True, "fallback": (env_fallback, ['IDRAC_X_AUTH_TOKEN'])},
     "idrac_port": {"required": False, "default": 443, "type": 'int'},
     "validate_certs": {"type": "bool", "default": True},
     "ca_path": {"type": "path"},
     "timeout": {"type": "int", "default": 30},
-
 }
+auth_mutually_exclusive = [["idrac_user", "x_auth_token"], ["idrac_password", "x_auth_token"]]
```
Comment (sachin-apa) [inline plugins/module_utils/idrac_redfish.py]:
```suggestion
auth_mutually_exclusive = [("idrac_user", "x_auth_token"), ("idrac_password", "x_auth_token")]
```
Can you check the syntax i think it should a list of tuple. similarly for other files.

File: plugins/module_utils/idrac_redfish.py [hunk 6]
```diff
@@ -41,14 +41,16 @@
...
+auth_required_together = [["idrac_user", "idrac_password"]]
```
Comment (sachin-apa) [inline plugins/module_utils/idrac_redfish.py]:
```suggestion
auth_required_together = [("idrac_user", "idrac_password")]
```
Can you check the syntax i think it should a list of tuple.  similarly for other files.

**Comment by rajshekarp87** [comment]
Update Version, copyright and python version in redfish_firmware

**Comment by jagadeeshnv** [comment]
Update the rst files

**Comment by sachin-apa** [comment]
This is not required all the changes has been taken care os part of the PR #662

**Pull request closed without merging**